src/Components/Dropdown.purs

```diff
@@ -0,0 +1,171 @@
+module Ocelot.Components.Dropdown where
+
+import Prelude
+
+import Data.Array (mapWithIndex)
+import Data.Maybe (Maybe(..), maybe)
+import Effect.Aff.Class (class MonadAff)
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Ocelot.Block.Button as Button
+import Ocelot.Block.Icon as Icon
+import Select as Select
+import Select.Utils.Setters (setContainerProps, setItemProps, setToggleProps)
+
+data Query item a
+  = HandleSelect (Select.Message (Query item) item) a
+  | Receive (Input item) a
+  | SetItems (Array item) a
+
+type State item =
+  { isOpen :: Boolean
+  , selectedItem :: Maybe item
+  , items :: Array item
+  , label :: String
+  , toString :: item -> String
+  , disabled :: Boolean
+  }
+
+type Input item =
+  { selectedItem :: Maybe item
+  , items :: Array item
+  , label :: String
+  , toString :: item -> String
+  , disabled :: Boolean
+  }
+
+data Message item = ItemSelected item
+
+type ChildSlot = Unit
+
+type ChildQuery item = Select.Query (Query item) item
+
+component
+  :: ∀ m item
+   . MonadAff m
+  => Eq item
+  => H.Component HH.HTML (Query item) (Input item) (Message item) m
+component =
+  H.parentComponent
+    { initialState
+    , render
+    , eval
+    , receiver: HE.input Receive
+    }
+
+  where
+    initialState :: Input item -> State item
+    initialState { selectedItem, items, label, toString, disabled } =
+      { isOpen: false
+      , selectedItem
+      , items
+      , label
+      , toString
+      , disabled
+      }
+
+    eval
+      :: Query item
+      ~> H.ParentDSL
+          (State item)
+          (Query item)
+          (ChildQuery item)
+          (ChildSlot)
+          (Message item)
+          m
+    eval = case _ of
+      HandleSelect message a -> case message of
+        Select.Selected item -> do
+          _ <- H.query unit $ Select.setVisibility Select.Off
+          H.raise $ ItemSelected item
+          H.modify_ _ { selectedItem = Just item }
+          pure a
+        _ -> pure a
+      Receive { selectedItem, items, label, disabled } a -> do
+        H.modify_ _
+          { selectedItem = selectedItem
+          , items = items
+          , label = label
+          , disabled = disabled }
+        pure a
+      SetItems items a -> do
+        H.modify_ _ { items = items }
+        pure a
+
+    render
```

Comment:
Are we going to have various dropdown styles? If so, it might make sense to make this component renderless and pass in the render function (like the typeahead). If not, we can leave it like this (much simpler) for the time being.

Reply:
There are currently two variants that mirror the buttons: light and dark. However the only difference is in the way the toggle is rendered, so I've just made it so that the component is parameterized by one of our button functions, and exported the two partially applied variants. It's possible that when we get to the point of writing the split button/dropdown behavior we may want to rip out the rendering entirely and use our `Store` method.